react.component/src/context/Context.tsx

```diff
@@ -1,51 +1,62 @@
 import { BookType } from '../app/types';
-import { createContext, ReactNode, useEffect, useReducer, useState } from 'react';
-
-interface CurrentType {
-  setIsLoading: (value: boolean)=> void;
-  isLoading: boolean;
-  docs: BookType[];
-}
+import { createContext, ReactNode, SetStateAction, useEffect, useState } from 'react';
 
 type ChildrenProps = {
   children: ReactNode;
 };
 
-export const AppContext = createContext<any>({
-  setIsLoading: () => {},
+type ContextType = {
+  isLoading: boolean;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setDocs: React.Dispatch<React.SetStateAction<BookType[]>>;
+  docs: BookType[];
+  error: string | null;
+  getDocs: () => Promise<BookType[]>;
+};
+
+export const AppContext = createContext<ContextType>({
   isLoading: true,
   docs: [],
+  setIsLoading: function (value: SetStateAction<boolean>): void {
+    throw new Error(`Function not implemented. ${value}`);
+  },
+  setDocs: function (value: SetStateAction<BookType[]>): void {
+    throw new Error(`Function not implemented. ${value}`);
+  },
+  error: null,
+  getDocs: function (): Promise<BookType[]> {
+    throw new Error('Function not implemented.');
+  },
 });
```

Comment:
![Безымянный](https://user-images.githubusercontent.com/88984698/230084098-a17cf99d-a890-47f0-8d54-099056cb2240.png)
i can't do this .... because of this error
![image](https://user-images.githubusercontent.com/88984698/230084248-01228156-c6a1-4fdc-b928-1ab59af5ebe0.png)